components/about-page-dropdown.js

```diff
@@ -0,0 +1,81 @@
+import { Disclosure } from '@headlessui/react';
+import { ChevronUpIcon } from '@heroicons/react/outline';
+
+export default function AboutPageDropdown({ currentSlug, slugs }) {
+  const remainingTitles = slugs?.filter((el) => el.slug !== currentSlug) ?? [];
+  const currentPage = slugs?.filter((el) => el.slug === currentSlug) ?? [];
+
+  return (
+    <div className='pt-8 w-full'>
+      <Disclosure>
+        {({ open }) => (
+          <>
+            <Disclosure.Button
+              className={`${
+                open ? 'rounded-t-md' : 'rounded-md'
+              } text-black bg-green-500 hover:text-green-500 flex justify-between items-center border border-green-500 h-10 w-full focus:outline-none focus-visible:ring focus-visible:ring-green-500 focus-visible:ring-opacity-75`}
+            >
+              <div
+                className={`${
+                  open ? 'rounded-tl-md' : 'rounded-l-md'
+                } h-full bg-white w-11/12 flex items-center justify-start pl-3 truncate`}
+              >
+                <span className='inline text-left '>{currentPage[0]?.title}</span>
+              </div>
+              <div className='w-1/12 px-5 h-full pr-5 bg-green-500 grid items-center justify-center rounded-r-md border border-green-500'>
+                <ChevronUpIcon
+                  className={`${open ? '' : 'rotate-180 transform'} h-5 w-5 text-white z-10`}
+                />
+              </div>
+            </Disclosure.Button>
+
+            <Disclosure.Panel>
+              <Disclosure.Button as='a' href={`/about/${remainingTitles[0]?.slug}`}>
```

Comment:
could we do` remainingTitles.map()` here?

Reply:
I totally agree and I tried that but for some reason it was not working :/. When I would open and close the drop-down there would just be an empty div when using map.

Reply:
strange!